### PR TITLE
chore: update package dependencies

### PR DESCRIPTION
**Pull request by keithamus**
This commit updates all dependencies. After this we should be able to get gk running properly - as I'm not sure it is right now.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,13 +53,13 @@
...
   "devDependencies": {
     "benchmark": "^2.1.0",
-    "browserify": "^13.0.0",
-    "browserify-istanbul": "^1.0.0",
-    "coveralls": "2.11.9",
-    "eslint": "^2.9.0",
-    "eslint-config-strict": "^8.5.0",
-    "eslint-plugin-filenames": "^0.2.0",
-    "ghooks": "^1.0.1",
+    "browserify": "^14.3.0",
```
Comment (keithamus) [inline package.json]: FYI, while Browserify `14` _technically_ no longer supports IE10, the main reason is due to it upgrading the `Buffer` module, which we don't make use of - so I feel like we can upgrade for this case - as it means less effort in trying to determine what packages need upgrading later on. However I also can see this is a slippery slope - as all we need to do is reference `Buffer` once to break browserify builds for IE10 - so I'm happy to take direction here.

Comment (lucasfcosta) [inline package.json]: I think that we won't need to reference Buffer anytime soon given the purpose of this library.
If we ever need to we can always go back to the old version.

I agree with this change. Also, thanks for explaining, it's always good to have this kind of comment available on GitHub if we ever need to see the reason why we're doing this.

File: package.json [hunk 2]
```diff
@@ -57,13 +53,13 @@
...
+    "browserify-istanbul": "^2.0.0",
+    "coveralls": "2.13.0",
+    "eslint": "^3.19.0",
+    "eslint-config-strict": "^13.0.0",
+    "eslint-plugin-filenames": "^1.1.0",
+    "husky": "^0.13.3",
```
Comment (lucasfcosta) [inline package.json]: I didn't know [this](https://www.npmjs.com/package/husky). Seems awesome!

Comment (keithamus) [inline package.json]: Yeah sorry - I should have mentioned; [ghooks is now deprecated](https://github.com/gtramontina/ghooks#deprecated) in favour of husky.

**Comment by keithamus** [comment]
Looks like the build is failing here because eslint no longer supports Node 0.10 or 0.12. We have PR #96 which should be merged first, and then this one will pass.

**Comment by lucasfcosta** [comment]
@keithamus, ah sorry, I didn't see that. Seems like an ultra easy issue to solve btw.

I already approved that PR.

**Pull request merged**